src/util/featureFlag.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * All available filter fields for the comments search form.
+ */
+export const COMMENT_SEARCH_FIELDS = [
+  "query",
+  "resultsPerPage",
+  "page",
+  "publishedAtFrom",
+  "publishedAtTo",
+  "sentiment",
+] as const;
+
+/**
+ * Filters out any fields that are listed in disabledFields.
+ * @param allFields - The complete list of fields.
+ * @param disabledFields - Fields to remove.
+ * @returns A subset of allFields excluding disabledFields.
+ */
+export const getEnabledFields = (
+  allFields: readonly string[],
+  disabledFields: string[],
+) => {
+  return allFields.filter((field) => !disabledFields.includes(field));
+};
+
+/**
+ * Reads the `COMMENT_FILTERING_DISABLED` env var (comma-separated list)
+ * and returns an array of disabled field names.
+ */
+export const disabledCommentSearchFields =
+  process.env.COMMENT_FILTERING_DISABLED?.split(",").filter(Boolean) ?? [];
+
+/**
+ * Computed list of enabled comment filter fields, based on feature flags.
+ */
+export const commentSearchFields = getEnabledFields(
+  COMMENT_SEARCH_FIELDS,
+  disabledCommentSearchFields,
+);
+
+if (process.env.NODE_ENV === "development") {
+  console.log("Disabled Comment Search Fields:", disabledCommentSearchFields);
+  console.log("Enabled Comment Search Fields:", commentSearchFields);
+}
```

Comment:
We should probably move these into a function so its called when we're directly using the functionality